command/host_create.go

```diff
@@ -109,58 +109,88 @@ func (c *createHost) Execute(ctx context.Context, comm client.Communicator,
 	return nil
 }
 
+type logBatchInfo struct {
+	hostID     string
+	outFile    *os.File
+	errFile    *os.File
+	batchStart time.Time
+	batchEnd   time.Time
+}
+
 func (c *createHost) getLogsFromNewDockerHost(ctx context.Context, logger client.LoggerProducer, comm client.Communicator,
 	ids []string, startTime time.Time, workDir string) error {
 	var err error
 	if len(ids) == 0 {
 		return errors.New("Programmer error: no intent host ID received")
 	}
 
+	info, err := c.initializeLogBatchInfo(ids[0], workDir, startTime)
```

Comment:
You should defer closing both files in this scope if foreground, or in the wait thread if background

Reply:
Ah i forgot about closing the files. Does this look okay?

Reply:
I'm ignoring the closing files errors since closing files to me feels like good style but not strictly necessary? I can use a catcher instead though.